### ISSUE TITLE
Rule proposal: Enforce type case (PascalCase)

### ISSUE THREAD

**Issue by fregante** [typescript-eslint/typescript-eslint#1882]
From what I see, type names are are usually in PascalCase. I think it would make sense to enforce this style.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Enforces code style (layout)
[ ] Other (please specify:)

## Fails

```ts
type boolFunction = () => boolean
```
```ts
interface someObject {}
```

## Passes

```ts
type BoolFunction = () => boolean
```
```ts
interface SomeObject {}
```

**Comment by bradzacher** [comment]
Have you seen the `naming-convention` rule? You can use the `typeLike` selector to match types.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md

**Comment by fregante** [comment]
Yep I think that’s it! I superficially searched for “case” among the rules and I hadn’t found it
